services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -124,11 +124,29 @@ function checkUnicity(primaryKeys, uniqueIndexes, columnName) {
       indexColumnName.length === 1 && indexColumnName.includes(columnName));
 
   const isPrimary = _.isEqual([columnName], primaryKeys);
-  return { isPrimary, isUnique };
+  return isPrimary || isUnique;
+}
+
+function isAssociationNameAlreadyExist(existingReferences, newReference) {
+  return existingReferences.some((reference) => reference.as === newReference.as);
+}
+
+function isReferenceAlreadyExist(existingReferences, newReference) {
+  return existingReferences.some((reference) => (
```

Comment:
Not fond of these names. They combine `is` and a verb.
I'd use either
* `associationNameAlreadyExists` or `associationNameIsAlreadyDefined` or `associationNameIsAlreadyUsed`.
* `referenceAlreadyExists` or `referenceIsAlreadyDefined` or `referenceIsAlreadyUsed`